cfep-13.md

```diff
@@ -0,0 +1,70 @@
+
+<table>
+<tr><td> Title </td><td> Securing `conda-forge` Uploads to `anaconda.org` </td>
+<tr><td> Status </td><td> Proposed </td></tr>
+<tr><td> Author(s) </td><td> Matt, Marius, Anthony, Isuru</td></tr>
+<tr><td> Created </td><td> Feb 27, 2020</td></tr>
+<tr><td> Updated </td><td> Feb 27, 2020</td></tr>
+<tr><td> Discussion </td><td> NA </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+Currently the `anaconda.org` tokens on each feedstock allow any `conda-forge`
+feedstock maintainer to upload any output to any `anaconda.org` package.
+This could allow someone to push a compromised and/or malicious build
+of say the `python` package. This situation is a security risk that we
+should mitigate.
+
+## Specification
+
+We will secure the uploads of outputs to `anaconda.org` via the following process.
+
+1. Each feedstock is provisioned with a secret key that is only known to the
+   maintainers of that feedstock and the core team. These keys will be kept in
+   a private repo with only core members having read access. Core members and 
+   encrypted GitHub keys for bot admins will have write access in order to generate 
+   the secret keys and push then to the CI providers.
+2. A public, global registry of all outputs from all feedstocks will be kept
+   in a repo on github. Write access to this repo is limited to core and bot 
+   admins as well.
+3. When a feedstock builds new outputs it wants to upload, it will push these outputs
+   to a staging organization on `anaconda.org`. It will also send the names of the
+   outputs, their SHA256 checksums, the feedstock name, and the secret key to a
+   validation web service. This web service then verifies the secret key-feedstock
+   combination and verifies that the outputs are allowed for that feedstock.
+4. If everything is ok, the outputs are copied from the staging organization to the
+   main conda-forge channels.
```

Comment:
We should probably delete bad/old artifacts from the staging channel after some period of time (28 days?).  I don't want the staging channel to be a usuable mirror of conda-forge, or else people could use that and it would be a security hole

Reply:
After copying, we could delete right?

Reply:
Yeah we can do that. But this is on them. They should not use an unsecured channel. We cannot control what they do.

Reply:
I think yeah we could delete after a copy

Reply:
done.